lib/openapi_parser/schema_validators/object_validator.rb

```diff
@@ -2,27 +2,47 @@ class OpenAPIParser::SchemaValidator
   class ObjectValidator < Base
     # @param [Hash] value
     # @param [OpenAPIParser::Schemas::Schema] schema
-    def coerce_and_validate(value, schema)
+    # @param [Boolean] parent_all_of true if component is nested under allOf
+    def coerce_and_validate(value, schema, parent_all_of: false)
       return OpenAPIParser::ValidateError.build_error_result(value, schema) unless value.kind_of?(Hash)
 
       return [value, nil] unless schema.properties
 
       required_set = schema.required ? schema.required.to_set : Set.new
+      remaining_keys = value.keys
 
       coerced_values = value.map do |name, v|
         s = schema.properties[name]
-        coerced, err = validatable.validate_schema(v, s)
+        coerced, err = if s
+                         remaining_keys.delete(name)
+                         validatable.validate_schema(v, s)
+                       else
+                         # Property is not defined, try validate using additionalProperties
+                         validate_using_additional_properties(schema, name, v)
+                       end
+
         return [nil, err] if err
 
         required_set.delete(name)
         [name, coerced]
       end
 
+      if !remaining_keys.empty? && !parent_all_of && !schema.additional_properties
+        # If object is nested in all of, the validation is already done in allOf validator. Or if
+        # additionalProperties are defined, we will validate using that
+        return [nil, OpenAPIParser::NotExistPropertyDefinition.new(remaining_keys, schema.object_reference)]
+      end
       return [nil, OpenAPIParser::NotExistRequiredKey.new(required_set.to_a, schema.object_reference)] unless required_set.empty?
 
       value.merge!(coerced_values.to_h) if @coerce_value
 
       [value, nil]
     end
+
+    def validate_using_additional_properties(schema, name, v)
+      # TODO: we need to perform a validation based on schema.additional_properties here
```

Comment:
I think we don't need implement this method in this PR.
Because this PR is for adding strict check feature and we can add this feature later.

So I will merge this and release a new version.
(If you have a problem, please tell me)

Reply:
ok, removing the method for now, but keeping the todo, so we know where to touch (and we can search for additionalProperties)